modules/google/etcd/ignition.tf

```diff
@@ -1,37 +1,86 @@
 data "ignition_config" "etcd" {
+  count = "${length(var.external_endpoints) == 0 ? var.instance_count : 0}"
+
   systemd = [
-    "${data.ignition_systemd_unit.locksmithd.id}",
-    "${data.ignition_systemd_unit.etcd3.id}",
+    "${data.ignition_systemd_unit.locksmithd.*.id[count.index]}",
+    "${data.ignition_systemd_unit.etcd3.*.id[count.index]}",
+    "${data.ignition_systemd_unit.etcd_unzip_tls.id}",
   ]
 
   files = [
-    "${data.ignition_file.node_hostname.id}",
+    "${data.ignition_file.node_hostname.*.id[count.index]}",
+    "${data.ignition_file.etcd_tls_zip.id}",
   ]
 }
 
 data "ignition_file" "node_hostname" {
+  count      = "${length(var.external_endpoints) == 0 ? var.instance_count : 0}"
   path       = "/etc/hostname"
   mode       = 0644
   filesystem = "root"
 
   content {
-    content = "${var.cluster_name}-etcd.${var.base_domain}"
+    content = "${var.cluster_name}-etcd-${count.index}.${var.base_domain}"
   }
 }
 
+data "ignition_file" "etcd_tls_zip" {
+  path       = "/etc/ssl/etcd/tls.zip"
+  mode       = 0400
+  uid        = 0
+  gid        = 0
+  filesystem = "root"
+
+  content {
+    mime    = "application/octet-stream"
+    content = "${var.tls_zip}"
+  }
+}
+
+data "ignition_systemd_unit" "etcd_unzip_tls" {
```

Comment:
this was added to AWS due to its userdata limit of 21k. Do we have the same limitations in GCP?

If not, I'd suggest to inject the certs as we do in i.e. Azure to reduce complexity.: https://github.com/coreos/tectonic-installer/blob/54be4a9/modules/azure/etcd/ignition.tf